### PR TITLE
Bot filtering implementation

### PR DESCRIPTION
**Pull request by AbdurRafay**

### CODE REVIEW

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m [hunk 1]
```diff
@@ -182,6 +183,22 @@ - (OPTLYAttribute *)getAttributeForKey:(NSString *)attributeKey {
     return attribute;
 }
 
+- (nullable NSString *)getAttributeIdForKey:(nonnull NSString *)attributeKey {
+    OPTLYAttribute *attribute = self.attributeKeyToAttributeMap[attributeKey];
```
Comment (msohailhussain) [inline OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m:187]: Make sure, code doesn't break here. if attributekey is not in the map.

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m [hunk 2]
```diff
@@ -182,6 +183,22 @@ - (OPTLYAttribute *)getAttributeForKey:(NSString *)attributeKey {
...
+    BOOL hasReservedPrefix = [attributeKey hasPrefix:kReservedAttributePrefix];
+    NSString *attributeId;
+    if (attribute) {
```
Comment (msohailhussain) [inline OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m:190]: One condition should be covered, if attributeKey is not reserved and not in datafile. Looks missing to me.

File: OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m (hunk 1 above)
Comment (AbdurRafay) [reply OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m:187]: Yes it is checked here https://github.com/optimizely/objective-c-sdk/blob/21f3a0cfa4055fd3eab43a6619c23585452dc883/OptimizelySDKCore/OptimizelySDKCore/OPTLYProjectConfig.m#L190

**Review by thomaszurkan-optimizely — CHANGES_REQUESTED** [review]
Can you fix the merge conflicts?  Also, why did code coverage drop 48%?

**Review by thomaszurkan-optimizely — APPROVED** [review]
LGTM

**Pull request merged**